internal/metrics/metrics.go

```diff
@@ -5,13 +5,117 @@ import (
 	"net/http"
 	"time"
 
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
 const (
 	MetricsEndpoint = "0.0.0.0:9090"
 )
 
+var (
+	EventsCounter *prometheus.CounterVec
+
+	ConditionCounter        *prometheus.CounterVec
+	ConditionRunTimeSummary *prometheus.SummaryVec
+
+	ActionCounter        *prometheus.CounterVec
+	ActionRuntimeSummary *prometheus.SummaryVec
+
+	ActionHandlerCounter        *prometheus.CounterVec
+	ActionHandlerRunTimeSummary *prometheus.SummaryVec
+
+	DownloadBytes          *prometheus.CounterVec
+	DownloadRunTimeSummary *prometheus.SummaryVec
+	UploadBytes            *prometheus.CounterVec
+	UploadRunTimeSummary   *prometheus.SummaryVec
+
+	StoreQueryErrorCount *prometheus.CounterVec
+)
+
+func init() {
+	EventsCounter = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "flasher_events_received",
+			Help: "A counter metric to measure the total count of events received",
+		},
+		[]string{"valid", "response"}, // valid is true/false, response is ack/nack
+	)
+
+	ConditionCounter = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "flasher_conditions_received",
+			Help: "A counter metric to measure the total count of condition requests received, successful and failed",
+		},
+		[]string{"condition", "state"},
```

Comment:
If `condition` is a condition ID, you will have a combinatoric explosion of time-series because the label is not bounded, and internally Prometheus will store a separate time-series for each label.

If it's a condition-type, this is flasher, do we expect that this would ever be something aside `installFirmware`?

Reply:
The value is a `ConditionType`, I was trying to follow a generic approach that other controllers can adopt/reference. For now though this will just be limited to the `installFirmware` condition.

Reply:
It feels like we are heading toward having a controller library or framework. If so we should probably make that intention explicit in the plan (i.e. in the Epic). That said, I think it’s a bit premature.

Reply:
I see your point here, let's discuss and go with the approach that works for both of us. I'm probably thinking ahead way too much.

Reply:
I agree that development strategy is out of scope for this pull-request. 😁 I'm OK with this as-is for now, but I do want to revisit it from a project-level perspective.